WalletWasabi/WabiSabi/Backend/Models/Alice.cs

```diff
@@ -40,5 +40,13 @@ private uint256 CalculateHash()
 				.Append(ProtocolConstants.AliceCoinOutpointStrobeLabel, Coin.Outpoint)
 				.Append(ProtocolConstants.AliceOwnershipProofStrobeLabel, OwnershipProof)
 				.GetHash();
+
+		public static uint256 CalculateHash(Coin coin, BitcoinSecret bitcoinSecret, uint256 roundId)
+		{
+			var ownershipProof = OwnershipProof.GenerateCoinJoinInputProof(
+				bitcoinSecret.PrivateKey,
+				new CoinJoinInputCommitmentData("CoinJoinCoordinatorIdentifier", roundId));
+			return new Alice(coin, ownershipProof).CalculateHash();
+		}
```

Comment:
I don't like reaching into the backend classes only for the purpose of generating the ID, something that both the client and the server have a concept of.

It seems like the cleanest way to handle this stuff where there's two somewhat overlapping implementations of each concept (`Alice`, `Bob`, `Round`, ...) with the current code layout is to have the shared code in the `WabiSabi.Models` namespace and have that true to its name (no side effects or state), and have `Backend` and `Client` each have its own variant (subclassing or embedding as a property are both valid IMO, depends on the situation)

Reply:
@molnard imo the non-static `CalculateHash` should use the static one. Then we can move this method to somewhere else if necessary.

Reply:
I moved it to AliceClient. That satisfies all conditions.